corporate/lib/stripe.py

```diff
@@ -1131,6 +1138,38 @@ def update_customer_minimum_licenses(self, new_minimum_license_count: int) -> st
             previous_minimum_license_count = 0
         return f"Minimum licenses for {self.billing_entity_display_name} changed to {new_minimum_license_count} from {previous_minimum_license_count}."
 
+    def set_required_plan_tier(self, required_plan_tier: int) -> str:
+        previous_required_plan_tier = None
+        new_plan_tier = None
+        if required_plan_tier != 0:
+            new_plan_tier = required_plan_tier
+        customer = self.get_customer()
+
+        if new_plan_tier is not None and not self.check_valid_plan_tier(required_plan_tier):
+            raise SupportRequestError(f"Invalid plan tier for {self.billing_entity_display_name}.")
+
+        if customer is not None:
+            previous_required_plan_tier = customer.required_plan_tier
+            customer.required_plan_tier = new_plan_tier
+            customer.save(update_fields=["required_plan_tier"])
+        else:
+            customer = self.update_or_create_customer(
+                defaults={"required_plan_tier": new_plan_tier}
+            )
+
+        self.write_to_audit_log(
+            event_type=AuditLogEventType.REQUIRED_PLAN_TIER_CHANGED,
+            event_time=timezone_now(),
+            extra_data={
+                "previous_required_plan_tier": previous_required_plan_tier,
+                "new_required_plan_tier": required_plan_tier,
+            },
+        )
```

Comment:
We probably should be using the pattern we use in `RealmAuditLog.REALM_PROPERTY_CHANGED` rather than making up a new event log type for every field with its own unique format for every `Customer` field changed. Maybe a `CUSTOMER_PROPERTY_CHANGED` audit log event type?

Reply:
Added an initial prep commit for this change. There's now a `CUSTOMER_PROPERTY_CHANGED` and `CUSTOMER_PLAN_PROPERTY_CHANGED` for the audit logs. These replace `CUSTOMER_MINIMUM_LICENSES_CHANGED` and `CUSTOMER_PLAN_END_DATE_CHANGED` respectively.

I'm not sure if we've set a customer.minimum_license or updated any plan.end_date since these were added in the past week. We'd just need to update the extra_data on the log for the new format.

Reply:
Yeah we can hand-fix this after this is merged, I think; very small numbers at least.